### PR TITLE
Development Documentation

### PR DESCRIPTION
**Pull request by SharonHart**
Add missing sections in the development guidelines.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,14 +21,6 @@ For more details follow the [Build and Release documentation](docs/build_release
 
 To get started, refer to the documentation for [setting up a development environment](docs/development.md).
 
-### General contribution guidelines
```
Comment (navalev) [inline CONTRIBUTING.md:24]: why was this removed?

File: docs/development.md [hunk 2]
```diff
@@ -6,15 +6,79 @@
 
 ### Cloning the repo
 
-TODO: Describe how to clone, folder structure etc.
+To create a local copy of Presidio repository, follow [Github instructions](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository)
+on how to clone a project using git.
+The project is structured so that:
+ - Each Preisio service has a designated directory. In it, the source code for its
```
Comment (navalev) [inline docs/development.md:12]:
```suggestion
 - Each Presidio service has a designated directory. In it, the source code for its
```

**Review by navalev — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (SharonHart) [reply CONTRIBUTING.md:24]: Moved to development.md, linked from here

**Review by navalev — APPROVED** [review]

File: docs/development.md [hunk 3]
```diff
@@ -6,15 +6,79 @@
 
 ### Cloning the repo
 
-TODO: Describe how to clone, folder structure etc.
+To create a local copy of Presidio repository, follow [Github instructions](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository)
+on how to clone a project using git.
+The project is structured so that:
+ - Each Presidio service has a designated directory. In it, the source code for its
+logic and tests, as well as the source code for serving it as an HTTP service and
```
Comment (shiranr) [inline docs/development.md:13]: I would consider changing this part a little to:
Each directory contains: 
- source code:
    - Engine - to use the service as python package
    - HTTP service - can be found in app.py/
- integration tests
- unit tests

File: docs/development.md [hunk 4]
```diff
@@ -6,15 +6,79 @@
...
+as a python package.
+ - In the project root directory, you will find common code for using, serving and testing Presidio
+as a cluster of services, as well as CI/CD pipelines codebase and documentation.
 
 ### Setting up Pipenv
 
-TODO: Add Pipenv documentation from V1.
+[Pipenv](https://pipenv.pypa.io/en/latest/) is a Python workflow manager, handling
+dependencies and environment for Python packages. It is used by each Presidio service
+as the dependencies manager, to avoid dependencies issues when switching between different
```
Comment (shiranr) [inline docs/development.md:22]: I think this sentence can be shorter something like: 
A dependency manager to align the versions.

File: docs/development.md [hunk 5]
```diff
@@ -6,15 +6,79 @@
...
+    ```
+
+3. Run all tests
+    ```
+    pipenv run pytest
+    ```
+
+4. To run arbitrary scripts within the virtual env, start the command with
+`pipenv run`. For example:
+    1. `pipenv run flake8"`
```
Comment (shiranr) [inline docs/development.md:55]: extra "

File: docs/development.md (hunk 4 above)
Comment (SharonHart) [reply docs/development.md:22]: Shortened

File: docs/development.md (hunk 3 above)
Comment (SharonHart) [reply docs/development.md:13]: Wanted a more holistic description here, so I don;t want to use 'engine' for example. Also not all have the same structure for tests, will try to make it more clear though

**Review by shiranr — APPROVED** [review]
